test: tests/group_by_test.py::test_group_by_shift_raises[cudf_constructor] fails

When running the tests with cuDF https://www.kaggle.com/code/marcogorelli/testing-cudf-in-narwhals?scriptVersionId=212927404 we have a test failure

```python
_________________ test_group_by_shift_raises[cudf_constructor] _________________

constructor = <function cudf_constructor at 0x78e02576c280>
request = <FixtureRequest for <Function test_group_by_shift_raises[cudf_constructor]>>

    @pytest.mark.filterwarnings("ignore:Found complex group-by expression:UserWarning")
    def test_group_by_shift_raises(
        constructor: Constructor, request: pytest.FixtureRequest
    ) -> None:
        if "polars" in str(constructor):
            # Polars supports all kinds of crazy group-by aggregations, so
            # we don't check that it errors here.
            request.applymarker(pytest.mark.xfail)
        df_native = {"a": [1, 2, 3], "b": [1, 1, 2]}
        df = nw.from_native(constructor(df_native))
        with pytest.raises(
            ValueError, match=".*(failed to aggregate|Non-trivial complex aggregation found)"
        ):
>           df.group_by("b").agg(nw.col("a").shift(1))

tests/group_by_test.py:359: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
narwhals/group_by.py:114: in agg
    self._grouped.agg(*aggs, **named_aggs),
narwhals/_pandas_like/group_by.py:101: in agg
    return agg_pandas(
narwhals/_pandas_like/group_by.py:300: in agg_pandas
    result_complex = grouped.apply(func)
/opt/conda/lib/python3.10/site-packages/cudf/utils/performance_tracking.py:51: in wrapper
    return func(*args, **kwargs)
/opt/conda/lib/python3.10/site-packages/cudf/core/groupby/groupby.py:1645: in apply
    if _can_be_jitted(grouped_values, func, args):
/opt/conda/lib/python3.10/site-packages/cudf/core/udf/groupby_utils.py:226: in _can_be_jitted
    _get_udf_return_type(dataframe_group_type, func, args)
/opt/conda/lib/python3.10/site-packages/cudf/utils/performance_tracking.py:51: in wrapper
    return func(*args, **kwargs)
/opt/conda/lib/python3.10/site-packages/cudf/core/udf/utils.py:98: in _get_udf_return_type
    ptx, output_type = cudautils.compile_udf(func, compile_sig)
/opt/conda/lib/python3.10/site-packages/cudf/utils/cudautils.py:119: in compile_udf
    key = make_cache_key(udf, type_signature)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

udf = <function agg_pandas.<locals>.func at 0x78e08c283490>
sig = (Record([('a', {'type': Group(int64, int64), 'offset': 0, 'alignment': None, 'title': None, }), ('b', {'type': Group(int64, int64), 'offset': 24, 'alignment': None, 'title': None, })], 48, True),)

    def make_cache_key(udf, sig):
        """
        Build a cache key for a user defined function. Used to avoid
        recompiling the same function for the same set of types
        """
        codebytes = udf.__code__.co_code
        constants = udf.__code__.co_consts
        names = udf.__code__.co_names
    
        if udf.__closure__ is not None:
            cvars = tuple(x.cell_contents for x in udf.__closure__)
>           cvarbytes = dumps(cvars)
E           AttributeError: Can't pickle local object 'reuse_series_implementation.<locals>.func'

/opt/conda/lib/python3.10/site-packages/cudf/utils/cudautils.py:80: AttributeError
```

cuDF raises on this operation, which is correct, but we should be raising a good error message instead of letting cuDF raise this

--- a/tests/group_by_test.py
+++ b/tests/group_by_test.py
@@ -346,6 +346,10 @@ def test_group_by_shift_raises(
         # Polars supports all kinds of crazy group-by aggregations, so
         # we don't check that it errors here.
         request.applymarker(pytest.mark.xfail)
+    if "cudf" in str(constructor):
+        # This operation fails completely in cuDF anyway, we just let raise its own
+        # error.
+        request.applymarker(pytest.mark.xfail)
     df_native = {"a": [1, 2, 3], "b": [1, 1, 2]}
     df = nw.from_native(constructor(df_native))
     with pytest.raises(

